### ISSUE TITLE
unable to catch pokemon

### ISSUE THREAD

**Issue by ramarro123** [Grover-c13/PokeGOAPI-Java#123]
After moving to the suggested api, i am unable to catch pokemon.

can someone explain the difference between
MapPokemon
and 
WildPokemon

tnx

**Comment by ramarro123** [comment]
ok, tested, i can confirm that

> ```
>     EncounterResponseOuterClass.EncounterResponse rsp = go.getMap().encounterPokemon(pokemon);
>     if (rsp.getStatus() == EncounterResponseOuterClass.EncounterResponse.Status.ENCOUNTER_SUCCESS) {
>         CatchPokemonResponseOuterClass.CatchPokemonResponse result = go.getMap().catchPokemon(pokemon, 1.0, 1.95 + Math.random() * 0.05, 0.85 + Math.random() * 0.15, ItemIdOuterClass.ItemId.ITEM_POKE_BALL_VALUE);
>         if (result.getStatus() == CatchPokemonResponseOuterClass.CatchPokemonResponse.CatchStatus.CATCH_SUCCESS) {
>             log("[+] catturato " + pokemon.getPokemonId());
>         } else {
>             logFile("non sono riuscito a catturare " + pokemon.getPokemonId());
> 
>         }
>     }
> ```

work (deprecated api)
while the new one, return always CATCH_ERROR

**Comment by sbley** [comment]
The error is in

```
public CatchResult catchPokemon() throws LoginFailedException, RemoteServerException {
        Pokeball ball = Pokeball.POKEBALL;
        if (api.getBag().getItem(ItemIdOuterClass.ItemId.ITEM_POKE_BALL).getCount() == 0) {
            ball = Pokeball.GREATBALL;
        }
        if (api.getBag().getItem(ItemIdOuterClass.ItemId.ITEM_GREAT_BALL).getCount() == 0) {
            ball = Pokeball.ULTRABALL;
        }
        if (api.getBag().getItem(ItemIdOuterClass.ItemId.ITEM_ULTRA_BALL).getCount() == 0) {
            ball = Pokeball.MASTERBALL;
        }
        return catchPokemon(ball);
    }
```

Imagine you have only Pokeball. Then the 2nd and 3rd if statement return true and a Masterball will be thrown. This results in a CATCH_ERROR.

**Comment by mjmjelde** [comment]
@sbley we have already identified this error and a PR will be submitted shortly for it

**Comment by Grover-c13** [comment]
Fixed in pr #129 closing
